src/udp_engine.cpp

```diff
@@ -111,9 +111,11 @@ void zmq::udp_engine_t::plug (io_thread_t *io_thread_, session_base_t *session_)
 
     if (send_enabled) {
         if (!options.raw_socket) {
-            out_address = address->resolved.udp_addr->dest_addr ();
-            out_addrlen = address->resolved.udp_addr->dest_addrlen ();
+            const ip_addr_t *out = address->resolved.udp_addr->target_addr ();
+            out_address = out->as_sockaddr ();
+            out_addrlen = out->sockaddr_len ();
         } else {
+            /// XXX fixme ?
             out_address = (sockaddr *) &raw_address;
             out_addrlen = sizeof (sockaddr_in);
```

Comment:
I don't quite understand how this `raw_address` part fits in, I see that the address is parsed within this very file in `out_event`, where does the string come from exactly? Why doesn't it use the `udp_address` parser?

Reply:
not sure about the address, but thw raw_socket option is used to read/write UDP datagrams without any ZMTP headers